### PR TITLE
magento/devdocs#:5185  Adding more information about configuring TinyMCE

### PR DESCRIPTION
**Pull request by DenisSaltanahmedov**
## Purpose of this pull request

This pull request (PR) adding more information about configuring TinyMCE topic

## Affected DevDocs pages
-  https://devdocs.magento.com/guides/v2.3/ui_comp_guide/components/wysiwyg/configure-tinymce-editor.html

## Links to Magento source code
- https://github.com/magento/devdocs/blob/master/src/guides/v2.3/ui_comp_guide/components/wysiwyg/configure-tinymce-editor.md

whatsnew
Big rewrite of the [Configure the TinyMCE editor](https://devdocs.magento.com/guides/v2.3/ui_comp_guide/components/wysiwyg/configure-tinymce-editor.html) topic.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by dobooth** [comment]
Good morning, @DenisSaltanahmedov I am also curious why so much information is being removed.

**Comment by DenisSaltanahmedov** [comment]
Hi, the PR is related to the issue https://github.com/magento/devdocs/issues/5185
According to an issue I adding more information about the TinyMCE Configuration.
Maybe I was mistaken with the requirements.
Thanks.

**Comment by rogyar** [comment]
Hi @dobooth. I believe that the initial version of the current document is not 100% correct. It explains how to extend WYSIWYG for the `PageBuilder` module. The `PageBuilder` module is a part of Magento Commerce only so far. 

At the same time, the current version of the document does not explain how to extend the configuration of the TinyMCE of the `Magento/Wysiwyg` module. 

So, the new version by @DenisSaltanahmedov brings information about extending the basic TinyMCE configuration (without dependencies with PageBuilder). Technically, this document is the right place for such information.

As far as I see, the current version of the document (without changes from @DenisSaltanahmedov) belongs rather to the [current](https://devdocs.magento.com/page-builder/docs/) section of the DevDocs related to the `PageBuilder` module itself. 

@DenisSaltanahmedov, please correct me if I'm wrong. If I'm not, please adjust the PR description correspondingly (see my [previous](https://github.com/magento/devdocs/pull/6681#pullrequestreview-361828292) comment for details).

Thank you.

**Comment by rogyar** [comment]
Hi @DenisSaltanahmedov. 

I agree, in this document the `PageBuilder` is being used as an example of TinyMCE extensibility. I'm not sure it's the best example since this document is not "Commerce-only". However, It's a good example. 

So, maybe we should not remove all the information about PageBuilder but extend this document with another example instead (and fix the existing misconceptions if any).

What do you think?

**Comment by DenisSaltanahmedov** [comment]
Hi, I have returned the PageBuilder example and added the example of CompositeConfigProvider, please check. Thanks.

**Comment by DenisSaltanahmedov** [comment]
Hi, @rogyar I have cleaned up the topic, could you review it.

**Comment by dobooth** [comment]
I was just wondering what the status of this was. Thanks for the update.

File: src/guides/v2.3/ui_comp_guide/components/wysiwyg/configure-tinymce-editor.md [hunk 1]
```diff
@@ -4,13 +4,61 @@ group: ui-components-guide
...
+        </argument>
+        <argument name="galleryConfigProvider" xsi:type="array">
+            <item name="Magento_PageBuilder/pageBuilderAdapter" xsi:type="string">Magento\Cms\Model\WysiwygDefaultConfig</item>
+        </argument>
+    </arguments>
+</type>
+```
 
+The class DefaultConfigProvider returns data required to render TinyMCE editor.
+From the example the class `Magento\PageBuilder\Model\Wysiwyg\DefaultConfigProvider` overrides existing configuration provided from the cms module.
```
Comment (rogyar) [inline src/guides/v2.3/ui_comp_guide/components/wysiwyg/configure-tinymce-editor.md]:
```suggestion
From the example the class `Magento\PageBuilder\Model\Wysiwyg\DefaultConfigProvider` overrides existing configuration provided by `Magento_CMS` module.
```

File: src/guides/v2.3/ui_comp_guide/components/wysiwyg/configure-tinymce-editor.md [hunk 2]
```diff
@@ -27,6 +75,82 @@ The following code shows an example of configuration settings in the `di.xml` fi
   </arguments>
 </type>
 ```
-See a list of possibly TinyMCE settings on their [website](https://www.tinymce.com/docs/configure/).
 
 Once you have edited the `di.xml` file you can apply custom styling for the settings you implemented in the related CSS file.
+
+Also, the config of the TinyMCE editor builds can be extended by a plugin.
```
Comment (rogyar) [inline src/guides/v2.3/ui_comp_guide/components/wysiwyg/configure-tinymce-editor.md]:
```suggestion
Also, the config of the TinyMCE editor can be extended by a plugin.
```

File: src/guides/v2.3/ui_comp_guide/components/wysiwyg/configure-tinymce-editor.md [hunk 3]
```diff
@@ -27,6 +75,82 @@ The following code shows an example of configuration settings in the `di.xml` fi
...
+Configuration providers are classes with a `getConfig()` method that returns the configuration for a specific entity.
+
+For example, in the `Magento\Cms\Model\Wysiwyg\DefaultConfigProvider` the method `getConfig`
```
Comment (rogyar) [inline src/guides/v2.3/ui_comp_guide/components/wysiwyg/configure-tinymce-editor.md]:
```suggestion
For example, in the `Magento\Cms\Model\Wysiwyg\DefaultConfigProvider` the method `getConfig()`
```

File: src/guides/v2.3/ui_comp_guide/components/wysiwyg/configure-tinymce-editor.md [hunk 4]
```diff
@@ -27,6 +75,82 @@ The following code shows an example of configuration settings in the `di.xml` fi
...
+                ),
+                'content_css' => $this->assetRepo->getUrl('mage/adminhtml/wysiwyg/tiny_mce/themes/ui.css')
+            ]
+        ]);
+        return $config;
+    }
+}
+```
+
+In the case the `Magento\Cms\Model\Wysiwyg\Gallery\DefaultConfigProvider` enables the image plugin.
```
Comment (rogyar) [inline src/guides/v2.3/ui_comp_guide/components/wysiwyg/configure-tinymce-editor.md]:
```suggestion
In this case, the `Magento\Cms\Model\Wysiwyg\Gallery\DefaultConfigProvider` enables the image plugin.
```

File: src/guides/v2.3/ui_comp_guide/components/wysiwyg/configure-tinymce-editor.md [hunk 5]
```diff
@@ -27,6 +75,82 @@ The following code shows an example of configuration settings in the `di.xml` fi
...
+See a list of possibly TinyMCE settings on their [website](https://www.tinymce.com/docs/configure/).
+
+In this case you can extend and fully customize the TinyMCE editor implementing `DefaultConfigProvider`, the class implements the method `getConfig` where specify necessary configuration. The method can by extended/overwrote of the after plugin.
```
Comment (rogyar) [inline src/guides/v2.3/ui_comp_guide/components/wysiwyg/configure-tinymce-editor.md]:
```suggestion
The following example shows how you can extend the TinyMCE editor configuration by creating a plugin for `DefaultConfigProvider` which extends  `getConfig()` method of the original class. 
```

**Comment by DenisSaltanahmedov** [comment]
Hi @rogyar, thank you for your help. I have fixed recommendations.

**Comment by dobooth** [comment]
running tests

**Pull request merged**